server.py

```diff
@@ -228,3 +231,17 @@ def semesterIDType(reqstr):
 
   # Convert Officer objects back to dictionaries before returning
   return jsonify([x.__dict__ for x in officers])
+
+calurl = 'https://calendar.google.com/calendar/ical/ca149os3pmnh0dcopr1jn2negg%40group.calendar.google.com/public/basic.ics'
+dtstrformat = '%Y-%m-%dT%H:%M:%S.%fZ'
+@app.route('/api/events')
+def getEvents():
+  gcal = Calendar.from_ical(requests.get(calurl).content)
```

Comment:
Should we be querying for the `.ics` file on every call, or should we do some form of caching and refresh it every so often?

Reply:
Yeah, it's probably best to cache it. How would you like to go about that?

Reply:
Dunno if there's a specific way in Python (maybe @wilt00 or @Benbigmac can answer that), but one way that I would do it is keep track of the last `.ics` response and the time that it was grabbed in some variable(s).

Then on every query to this endpoint...

* if the time difference between now and the request time is over a certain threshold (say 1 hour), grab a new copy of the `.ics` file, update the variable with the new `.ics` response a well as the request time (i.e. update the cache)
* return the value in the `.ics` variable

Reply:
What if I instead store the parsed list of events in cache? It would make calls within the cache period faster as the function would only have to jsonify the list in that situation.

Reply:
That does sound better; it'd limit the number of operations per successive call to the API endpoint.